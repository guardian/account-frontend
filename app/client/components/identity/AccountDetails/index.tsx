import Raven from "raven-js";
import React, { useEffect, useState } from "react";
import { headline } from "../../../styles/fonts";
import { trackEvent } from "../../analytics";
import { MembershipLinks } from "../../membershipLinks";
import { navLinks } from "../../nav";
import {
  PageContainer,
  PageHeaderContainer,
  PageNavAndContentContainer
} from "../../page";
import { Spinner } from "../../spinner";
import {
  GenericErrorMessage,
  GenericErrorMessageRef
} from "../GenericErrorMessage";
import { Users } from "../identity";
import { User } from "../models";
import { textSmall } from "../sharedStyles";
import { AccountDetailsFormSection } from "./AccountFormSection";

const errorRef = React.createRef<GenericErrorMessageRef>();
const pageTopRef = React.createRef<HTMLDivElement>();

const loader = (
  <PageContainer>
    <Spinner loadingMessage="Loading your profile ..." />
  </PageContainer>
);

export const AccountDetails = (props: { path?: string }) => {
  const [user, setUser] = useState();
  const [loading, setLoading] = useState(true);
  const [error, setError] = useState();
  const [emailMessage, setEmailMessage] = useState();

  useEffect(() => {
    if (error && errorRef.current) {
      window.scrollTo(0, errorRef.current.offsetTop - 20);
    }
  }, [error]);

  const handleGeneralError = (e: any) => {
    setError(true);
    Raven.captureException(e);
    trackEvent({
      eventCategory: "publicProfileError",
      eventAction: "error",
      eventLabel: e.toString()
    });
  };

  useEffect(() => {
    Users.getCurrentUser()
      .then((u: User) => {
        setUser(u);
      })
      .then(() => setLoading(false))
      .catch(handleGeneralError);
  }, []);

  const saveUser = async (values: User) => {
    const changedUser = { ...user, ...values };
    return await Users.saveChanges(user, changedUser);
  };

  const scrollToTop = () => {
    if (pageTopRef.current) {
      window.scrollTo(0, pageTopRef.current.offsetTop - 20);
    }
  };

  const updateValues = (input: User, response: User) => {
    const changedFields = Users.getChangedFields(response, input);
    if (changedFields.primaryEmailAddress) {
      setEmailMessage(changedFields.primaryEmailAddress);
    }
    setUser(response);
  };

  const content = () => (
    <>
      <div ref={pageTopRef} css={{ display: "none" }} />
      <PageContainer>
        <span css={textSmall}>
          These details will only be visible to you and the Guardian.
        </span>
      </PageContainer>
      <PageContainer>
        <AccountDetailsFormSection
          user={user}
          saveUser={saveUser}
          onError={handleGeneralError}
          onSuccess={updateValues}
          onDone={scrollToTop}
          emailMessage={emailMessage}
        />
      </PageContainer>
      <PageContainer>
        <MembershipLinks />
      </PageContainer>
    </>
  );

  return (
    <>
      <PageHeaderContainer selectedNavItem={navLinks.accountDetails}>
        <h1
          css={{
            fontSize: "32px",
            lineHeight: "36px",
            fontFamily: headline,
            marginBottom: "30px",
            marginTop: "0"
          }}
        >
          Account details
        </h1>
      </PageHeaderContainer>
      <PageNavAndContentContainer selectedNavItem={navLinks.accountDetails}>
        {!error || (
          <PageContainer>
            <GenericErrorMessage ref={errorRef} />
          </PageContainer>
        )}
        {loading ? loader : content()}
      </PageNavAndContentContainer>
    </>
  );
};
