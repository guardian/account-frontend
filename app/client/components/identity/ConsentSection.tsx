import React, { FC } from "react";
import { Consent } from "./identity";
import { MarketingPreference } from "./MarketingPreference";
import { PageSection } from "./PageSection";

type ClickHandler = (id: string) => {};

interface ConsentSectionProps {
  clickHandler: ClickHandler;
  consents: Consent[];
}

const consentPreference = (consent: Consent, clickHandler: ClickHandler) => {
  const { id, name, description } = consent;
  return (
    <MarketingPreference
      id={id}
      key={id}
      title={name}
      description={description}
      onClick={clickHandler}
    />
  );
};

const consentPreferences = (consents: Consent[], clickHandler: ClickHandler) =>
  consents.map(consent => consentPreference(consent, clickHandler));

export const ConsentSection: FC<ConsentSectionProps> = props => {
  const { consents, clickHandler } = props;
  return (
    <PageSection
      title="What else would you like to hear about by email?"
      description={`
        From time to time, we'd love to be able to send you information about
        our products, services and events.
      `}
    >
      {consentPreferences(consents, clickHandler)}
    </PageSection>
  );
};
