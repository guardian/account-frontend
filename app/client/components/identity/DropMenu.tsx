import React, { FC, useState } from "react";

interface DropMenuProps {
  title: string;
}

export const DropMenu: FC<DropMenuProps> = props => {
  const { children, title } = props;
  const [open, setOpen] = useState(false);
  return (
    <>
      <div className={open ? "open" : undefined} onClick={() => setOpen(!open)}>
        {title}
      </div>
      {open ? children : null}
    </>
  );
};
