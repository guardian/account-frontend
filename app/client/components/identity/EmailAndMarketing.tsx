import React from "react";
import { headline } from "../../styles/fonts";
import { navLinks } from "../nav";
import { PageContainer, PageHeaderContainer } from "../page";

export const EmailAndMarketing = (props: { path?: string }) => {
  return (
    <>
      <PageHeaderContainer selectedNavItem={navLinks.emailPrefs}>
        <h1
          css={{
            fontSize: "2rem",
            lineHeight: "2.25rem",
            fontFamily: headline,
            marginBottom: "30px",
            marginTop: "0"
          }}
        >
          Edit your profile
        </h1>
      </PageHeaderContainer>
      <PageContainer>
        <p>Email editing</p>
      </PageContainer>
    </>
  );
};
