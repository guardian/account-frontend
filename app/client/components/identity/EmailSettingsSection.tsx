import React, { FC } from "react";
import { Button } from "../buttons";
import { PageSection } from "./PageSection";

interface EmailSettingsSectionProps {
  email: string;
}

export const EmailSettingsSection: FC<EmailSettingsSectionProps> = props => {
  const { email } = props;
  return (
    <PageSection title={"Email settings"}>
      <p>
        You are recieving newsletters, notifications an all other emails to{" "}
        <strong>{email}</strong>
      </p>
      <p>Change your email address</p>
      <p>Manage your Jobs alerts</p>
      <Button text="Unsubscribe from all emails" />
    </PageSection>
  );
};
