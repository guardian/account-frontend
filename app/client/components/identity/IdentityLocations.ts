import { conf } from "../../../server/config";

const url = (subdomain: string, domain: string, path?: string) =>
  `https://${subdomain}.${domain}${path ? path : ""}`;

const DOMAIN =
  typeof window !== "undefined" && window.guardian
    ? window.guardian.domain
    : conf.DOMAIN;

const IMAGE_DOMAIN =
  DOMAIN === "theguardian.com" ? "guim.co.uk" : "guimcode.co.uk";

const IDAPI_URL =
  DOMAIN === "thegulocal.com" ? "/idapicodeproxy" : url("idapi", DOMAIN);

const AVATAR_URL =
  DOMAIN === "thegulocal.com" ? "/avatarcodeproxy" : url("avatar", DOMAIN);

const getIdentityLocations = (domain: string) => ({
  COMMUNITY_FAQS: url("www", domain, "/community-faqs"),
  CHANGE_EMAIL: url("profile", domain, "/account/edit"),
  MANAGE_JOB_ALERTS: url("jobs", domain, "/your-jobs/?ActiveSection=JbeList"),
  VERIFY_EMAIL: url("profile", domain, "/verify-email"),
  IDAPI: IDAPI_URL,
  AVATAR: AVATAR_URL,
  AVATAR_USER_IMAGES: url("avatar", IMAGE_DOMAIN, "/user")
});

export const IdentityLocations = getIdentityLocations(DOMAIN);
