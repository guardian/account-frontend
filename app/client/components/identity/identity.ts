import palette from "../../colours";

enum Theme {
  news = "news",
  features = "features",
  sport = "sport",
  culture = "culture",
  lifestyle = "lifestyle",
  comment = "comment",
  work = "work",
  FromThePapers = "From the papers"
}

export interface Consent {
  id: string;
  name: string;
  description: string;
}

interface ExactTargetEntity {
  exactTargetListId: number;
}

export interface Newsletter {
  id: string;
  theme: string;
  name: string;
  description: string;
  frequency: string;
}

export interface NewsletterGroup {
  theme: Theme;
  color: string;
  newsletters: Newsletter[];
}

const colors: { [T in Theme]: string } = {
  [Theme.news]: palette.red.medium,
  [Theme.features]: palette.neutral["1"],
  [Theme.sport]: palette.blue.medium,
  [Theme.culture]: "#a1845c",
  [Theme.lifestyle]: palette.pink.medium,
  [Theme.comment]: "#e05e00",
  [Theme.work]: palette.neutral["1"],
  [Theme.FromThePapers]: palette.neutral["1"]
};

// @TODO: DEV: POTENTIALLY REPLACEABLE IF CALLS PROXIED
const toNewsletter = (
  rawNewsletter: Newsletter & ExactTargetEntity
): Newsletter => {
  const {
    theme,
    name,
    description,
    frequency,
    exactTargetListId
  } = rawNewsletter;
  return {
    id: exactTargetListId.toString(),
    description,
    theme,
    name,
    frequency
  };
};

// @TODO: DEV: POTENTIALLY REPLACEABLE IF CALLS PROXIED
const toNewsletterGroups = (
  newsletters: Array<Newsletter & ExactTargetEntity>
): NewsletterGroup[] => {
  const template = [
    Theme.news,
    Theme.features,
    Theme.sport,
    Theme.culture,
    Theme.lifestyle,
    Theme.comment,
    Theme.work,
    Theme.FromThePapers
  ];
  return template.map(theme => ({
    theme,
    color: colors[theme],
    newsletters: newsletters
      .filter(newsletter => newsletter.theme === theme)
      .map(toNewsletter)
  }));
};

// @TODO: DEV: TESTING FUNCTION
export const updateNewsletter = async (
  id: string,
  subscribed: boolean = true
) => {
  const url = "https://idapi.thegulocal.com/users/me/newsletters";
  const payload = {
    id,
    subscribed
  };
  const options = {
    method: "PATCH",
    headers: {
      "Content-Type": "application/json"
    },
    body: JSON.stringify(payload)
  };
  const response = await fetch(url, options);
  if (!response.ok) {
    throw Error(
      "This is a test function and should NOT be present in the final merge: Failed to update newsletter data"
    );
  } else {
    return;
  }
};

// @TODO: DEV: TESTING FUNCTION
export const readNewsletters = async (): Promise<NewsletterGroup[]> => {
  const url = "https://idapi.thegulocal.com/newsletters";
  const response = await fetch(url);
  if (!response.ok) {
    throw Error(
      "This is a test function and should NOT be present in the final merge: Failed to retrieve newsletter data"
    );
  } else {
    return toNewsletterGroups(await response.json());
  }
};

// @TODO: DEV: POTENTIALLY REPLACEABLE IF CALLS PROXIED
export const mapConsentGroup = (consents: Consent[]): Consent[] => {
  const template = ["supporter", "jobs", "holidays", "events", "offers"];
  return template.reduce((consentGroup: Consent[], id) => {
    const consent = consents.find(c => c.id === id);
    if (consent) {
      consentGroup.push(consent);
    }
    return consentGroup;
  }, []);
};

// @TODO: DEV: TESTING FUNCTION
export const updateConsent = async (id: string, consented: boolean = true) => {
  const url = "https://idapi.thegulocal.com/users/me/consents";
  const payload = {
    id,
    consented
  };
  const options = {
    method: "PATCH",
    headers: {
      "Content-Type": "application/json"
    },
    body: JSON.stringify(payload)
  };
  const response = await fetch(url, options);
  if (!response.ok) {
    throw Error(
      "This is a test function and should NOT be present in the final merge: Failed to update consent data"
    );
  } else {
    return;
  }
};

// @TODO: DEV: TESTING FUNCTION
export const readConsents = async (): Promise<Consent[]> => {
  const url = "https://idapi.thegulocal.com/consents";
  const response = await fetch(url);
  if (!response.ok) {
    throw Error(
      "This is a test function and should NOT be present in the final merge: Failed to retrieve consents data"
    );
  } else {
    return await response.json();
  }
};
