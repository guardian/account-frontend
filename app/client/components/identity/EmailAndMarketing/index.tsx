import React, { useEffect, useState } from "react";
import palette from "../../../colours";
import { headline } from "../../../styles/fonts";
import { MembershipLinks } from "../../membershipLinks";
import { navLinks } from "../../nav";
import { PageContainer, PageHeaderContainer } from "../../page";
import { Spinner } from "../../spinner";
import { ConsentOptions, Users } from "../identity";
import { IdentityLocations } from "../IdentityLocations";
import { Lines } from "../Lines";
import { MarginWrapper } from "../MarginWrapper";
import { Actions, useConsentOptions } from "../useConsentOptions";
import { ConsentSection } from "./ConsentSection";
import { EmailSettingsSection } from "./EmailSettingsSection";
import { NewsletterSection } from "./NewsletterSection";
import { OptOutSection } from "./OptOutSection";

export const EmailAndMarketing = (props: { path?: string }) => {
  const { options, error, subscribe, unsubscribe, unsubscribeAll } = Actions;
  const [email, setEmail] = useState();
  const [removed, setRemoved] = useState(false);
  const [state, dispatch] = useConsentOptions();

  const toggleSubscription = async (id: string) => {
    const option = ConsentOptions.findById(state.options, id);
    try {
      if (option === undefined) {
        throw Error("Id not found");
      }
      if (option.subscribed) {
        await ConsentOptions.unsubscribe(option);
        dispatch(unsubscribe(id));
      } else {
        await ConsentOptions.subscribe(option);
        dispatch(subscribe(id));
      }
    } catch (e) {
      dispatch(error(e));
    }
  };

  const setRemoveAllEmailConsents = async () => {
    try {
      await ConsentOptions.unsubscribeAll();
      setRemoved(true);
      dispatch(unsubscribeAll());
    } catch (e) {
      dispatch(error(e));
    }
  };

  useEffect(() => {
    const makeInitialAPICalls = async () => {
      try {
        const user = await Users.getCurrentUser();
        if (!user.validated) {
          window.location.assign(IdentityLocations.VERIFY_EMAIL);
          return;
        }
        const consentOptions = await ConsentOptions.getAll();
        setEmail(user.email);
        dispatch(options(consentOptions));
      } catch (e) {
        dispatch(error(e));
      }
    };
    makeInitialAPICalls();
  }, []);

  const newsletters = ConsentOptions.newsletters(state.options);
  const consents = ConsentOptions.consents(state.options);
  const loading = newsletters.length === 0 && consents.length === 0;

  const errorRef = React.createRef<HTMLDivElement>();

  useEffect(
    () => {
      if (state.error && errorRef.current) {
        window.scrollTo(0, errorRef.current.offsetTop - 20);
      }
    },
    [state.error]
  );

  const errorMessage = (
    <PageContainer>
      <div
        ref={errorRef}
        css={{
          fontSize: "13px",
          lineHeight: "18px",
          backgroundColor: "#ffe1e1",
          borderBottom: `1px solid ${palette.red.light}`,
          borderTop: `1px solid ${palette.red.light}`,
          color: palette.red.medium,
          marginTop: "6px",
          padding: "7px 8px"
        }}
      >
        <p css={{ marginBottom: "10px" }}>Sorry, something went wrong!</p>
        <p css={{ marginBottom: "0" }}>
          <a
            css={{
              color: palette.blue.dark,
              cursor: "pointer"
            }}
            onClick={() => window.location.reload()}
          >
            Refresh this page
          </a>
        </p>
      </div>
    </PageContainer>
  );

  const content = (
    <>
      <PageContainer>
        <NewsletterSection
          newsletters={newsletters}
          clickHandler={toggleSubscription}
        />
      </PageContainer>
      <PageContainer>
        <MarginWrapper>
          <Lines n={4} />
        </MarginWrapper>
      </PageContainer>
      <PageContainer>
        <ConsentSection consents={consents} clickHandler={toggleSubscription} />
      </PageContainer>
      <PageContainer>
        <MarginWrapper>
          <Lines n={1} />
        </MarginWrapper>
      </PageContainer>
      <PageContainer>
        <EmailSettingsSection
          email={email}
          actionHandler={setRemoveAllEmailConsents}
          removed={removed}
        />
      </PageContainer>
      <PageContainer>
        <MarginWrapper>
          <Lines n={4} />
        </MarginWrapper>
      </PageContainer>
      <PageContainer>
        <OptOutSection consents={consents} clickHandler={toggleSubscription} />
      </PageContainer>
      <PageContainer>
        <MarginWrapper>
          <MembershipLinks />
        </MarginWrapper>
      </PageContainer>
    </>
  );

  const loader = (
    <PageContainer>
      <Spinner loadingMessage="Loading your email preferences..." />
    </PageContainer>
  );

  return (
    <>
      <PageHeaderContainer selectedNavItem={navLinks.emailPrefs}>
        <h1
          css={{
            fontSize: "32px",
            lineHeight: "36px",
            fontFamily: headline,
            marginBottom: "30px",
            marginTop: "0"
          }}
        >
          Edit your profile
        </h1>
      </PageHeaderContainer>
      {state.error ? errorMessage : null}
      {loading ? (!state.error ? loader : null) : content}
    </>
  );
};
