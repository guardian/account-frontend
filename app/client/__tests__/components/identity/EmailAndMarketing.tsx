import React from "react";
import renderer from "react-test-renderer";
import { EmailAndMarketing } from "../../../components/identity/EmailAndMarketing";

describe("Email and marketing tab", () => {
  test("The page renders correctly", () => {
    const rendered = renderer.create(<EmailAndMarketing />);
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
