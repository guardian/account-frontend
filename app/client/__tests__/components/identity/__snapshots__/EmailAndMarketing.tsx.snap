// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Email and marketing tab The page renders correctly 1`] = `
Array [
  <div
    css={
      Object {
        "background": "#f6f6f6",
        "borderBottom": "1px solid #dcdcdc",
        "marginLeft": "auto",
        "marginRight": "auto",
        "paddingTop": "0.3125rem",
      }
    }
  >
    <div
      css={
        Object {
          "@media (min-width: 740px)": Object {
            "paddingLeft": "1.25rem",
            "paddingRight": "1.25rem",
          },
          "margin": "auto",
          "maxWidth": "980px",
          "paddingLeft": "0.625rem",
          "paddingRight": "0.625rem",
        }
      }
    >
      <h1
        css={
          Object {
            "fontFamily": "GH Guardian Headline, \\"GuardianTextEgyptian\\", Georgia, serif",
            "fontSize": "2rem",
            "lineHeight": "2.25rem",
            "marginBottom": "30px",
            "marginTop": "0",
          }
        }
      >
        Edit your profile
      </h1>
      <ul
        css={
          Object {
            "map": undefined,
            "name": "1oc9fqg",
            "next": undefined,
            "styles": "width:100%;position:relative;margin:0;padding:0;border-bottom:0;list-style-type:none;z-index:2;table-layout:fixed;grid-template-columns:repeat(auto-fit, minmax(40%, 1fr));display:grid;border-right:1px solid #dcdcdc;@media (min-width: 980px){grid-template-columns:repeat(auto-fit, minmax(7.5rem, 1fr));}",
          }
        }
        role="tablist"
      >
        <li
          css={
            Object {
              "@media (min-width: 740px)": Object {
                "minWidth": "155.5px",
              },
              "background": "0.0625rem solid #dcdcdc",
              "borderLeft": "1px solid #dcdcdc",
              "borderTop": "1px solid #dcdcdc",
              "display": "table-cell",
              "margin": 0,
              "width": "100%",
            }
          }
        >
          <a
            css={
              Object {
                "map": undefined,
                "name": "17lc5kn",
                "next": undefined,
                "styles": "font-size:1rem;font-weight:500;line-height:1.25rem;font-family:\\"GuardianTextEgyptian\\", Georgia, serif;display:block;box-sizing:border-box;min-height:36px;padding:4px 0 0 5px;letter-spacing:-0.02rem;text-align:left;text-decoration:none;overflow:hidden;white-space:nowrap;text-overflow:ellipsis;background:;color:inherit;@media (min-width: 980px){box-shadow:;}",
              }
            }
            href="https://profile.thegulocal.com/public/edit"
          >
            Public profile
          </a>
        </li>
        <li
          css={
            Object {
              "@media (min-width: 740px)": Object {
                "minWidth": "155.5px",
              },
              "background": "0.0625rem solid #dcdcdc",
              "borderLeft": "1px solid #dcdcdc",
              "borderTop": "1px solid #dcdcdc",
              "display": "table-cell",
              "margin": 0,
              "width": "100%",
            }
          }
        >
          <a
            css={
              Object {
                "map": undefined,
                "name": "17lc5kn",
                "next": undefined,
                "styles": "font-size:1rem;font-weight:500;line-height:1.25rem;font-family:\\"GuardianTextEgyptian\\", Georgia, serif;display:block;box-sizing:border-box;min-height:36px;padding:4px 0 0 5px;letter-spacing:-0.02rem;text-align:left;text-decoration:none;overflow:hidden;white-space:nowrap;text-overflow:ellipsis;background:;color:inherit;@media (min-width: 980px){box-shadow:;}",
              }
            }
            href="https://profile.thegulocal.com/account/edit"
          >
            Account details
          </a>
        </li>
        <li
          css={
            Object {
              "@media (min-width: 740px)": Object {
                "minWidth": "155.5px",
              },
              "background": "0.0625rem solid #dcdcdc",
              "borderLeft": "1px solid #dcdcdc",
              "borderTop": "1px solid #dcdcdc",
              "display": "table-cell",
              "margin": 0,
              "width": "100%",
            }
          }
        >
          <a
            css={
              Object {
                "map": undefined,
                "name": "17lc5kn",
                "next": undefined,
                "styles": "font-size:1rem;font-weight:500;line-height:1.25rem;font-family:\\"GuardianTextEgyptian\\", Georgia, serif;display:block;box-sizing:border-box;min-height:36px;padding:4px 0 0 5px;letter-spacing:-0.02rem;text-align:left;text-decoration:none;overflow:hidden;white-space:nowrap;text-overflow:ellipsis;background:;color:inherit;@media (min-width: 980px){box-shadow:;}",
              }
            }
            href="/membership"
            onClick={[Function]}
          >
            Membership
          </a>
        </li>
        <li
          css={
            Object {
              "@media (min-width: 740px)": Object {
                "minWidth": "155.5px",
              },
              "background": "0.0625rem solid #dcdcdc",
              "borderLeft": "1px solid #dcdcdc",
              "borderTop": "1px solid #dcdcdc",
              "display": "table-cell",
              "margin": 0,
              "width": "100%",
            }
          }
        >
          <a
            css={
              Object {
                "map": undefined,
                "name": "17lc5kn",
                "next": undefined,
                "styles": "font-size:1rem;font-weight:500;line-height:1.25rem;font-family:\\"GuardianTextEgyptian\\", Georgia, serif;display:block;box-sizing:border-box;min-height:36px;padding:4px 0 0 5px;letter-spacing:-0.02rem;text-align:left;text-decoration:none;overflow:hidden;white-space:nowrap;text-overflow:ellipsis;background:;color:inherit;@media (min-width: 980px){box-shadow:;}",
              }
            }
            href="/subscriptions"
            onClick={[Function]}
          >
            Subscriptions
          </a>
        </li>
        <li
          css={
            Object {
              "@media (min-width: 740px)": Object {
                "minWidth": "155.5px",
              },
              "background": "0.0625rem solid #dcdcdc",
              "borderLeft": "1px solid #dcdcdc",
              "borderTop": "1px solid #dcdcdc",
              "display": "table-cell",
              "margin": 0,
              "width": "100%",
            }
          }
        >
          <a
            css={
              Object {
                "map": undefined,
                "name": "17lc5kn",
                "next": undefined,
                "styles": "font-size:1rem;font-weight:500;line-height:1.25rem;font-family:\\"GuardianTextEgyptian\\", Georgia, serif;display:block;box-sizing:border-box;min-height:36px;padding:4px 0 0 5px;letter-spacing:-0.02rem;text-align:left;text-decoration:none;overflow:hidden;white-space:nowrap;text-overflow:ellipsis;background:;color:inherit;@media (min-width: 980px){box-shadow:;}",
              }
            }
            href="/contributions"
            onClick={[Function]}
          >
            Contributions
          </a>
        </li>
        <li
          css={
            Object {
              "@media (min-width: 740px)": Object {
                "minWidth": "155.5px",
              },
              "background": "0.0625rem solid #ffffff",
              "borderLeft": "1px solid #dcdcdc",
              "borderTop": "1px solid #dcdcdc",
              "display": "table-cell",
              "margin": 0,
              "width": "100%",
            }
          }
        >
          <a
            css={
              Object {
                "map": undefined,
                "name": "160sf8k",
                "next": undefined,
                "styles": "font-size:1rem;font-weight:500;line-height:1.25rem;font-family:\\"GuardianTextEgyptian\\", Georgia, serif;display:block;box-sizing:border-box;min-height:36px;padding:4px 0 0 5px;letter-spacing:-0.02rem;text-align:left;text-decoration:none;overflow:hidden;white-space:nowrap;text-overflow:ellipsis;background:#ffffff;color:inherit;@media (min-width: 980px){box-shadow:0 1px 0 white;}",
              }
            }
            href="/email-prefs"
            onClick={[Function]}
          >
            Emails & marketing
          </a>
        </li>
      </ul>
    </div>
  </div>,
  <div
    css={
      Object {
        "@media (min-width: 740px)": Object {
          "padding": "0 1.25rem",
        },
        "margin": "1.8125rem auto 0",
        "maxWidth": "980px",
        "padding": "0 0.625rem",
      }
    }
  >
    <p>
      Email editing
    </p>
  </div>,
]
`;
